yt_dlp/extractor/nrk.py

```diff
@@ -457,11 +558,8 @@ def _real_extract(self, url):
 
         webpage = self._download_webpage(url, display_id)
 
-        info = self._search_json_ld(webpage, display_id, default={})
-        nrk_id = info.get('@id') or self._html_search_meta(
-            'nrk:program-id', webpage, default=None) or self._search_regex(
-            rf'data-program-id=["\']({NRKTVIE._EPISODE_RE})', webpage,
-            'nrk id')
+        info = self._parse_json(self._search_regex(r'<script id="pageData" type="application/json">[^{]*([^<]+)', webpage, 'page data json', group=1), '')
```

Comment:
Not sure what you mean by "hydration data", but yeah I think I see what you mean. That dict will contain a lot of data in a structure the extractor does not expect. I wasn't sure what the original `info` variable represented and was pretty much just trying to stop the test case from failing at this point.
And yeah, I think you're right. The URL in the test case now redirects to a URL covered by another extractor, so it's probably fine to just remove it. However, I still want to include the URL matching from this extractor for backwards compatibility since the `/sesong/1/episode/1` URL structure still resolves through redirects.

Reply:
>However, I still want to include the URL matching from this extractor for backwards compatibility since the `/sesong/1/episode/1` URL structure still resolves through redirects.

With an adjustment to `NRKTVSeasonIE` and `NRKTVSeriesIE` the generic extractor could handle the redirect.